src/helpers/loadOptimize.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { receiveExperiments } from '../app/featureFlags/actions';
+import { Store } from '../app/types';
+import { assertDocument, assertWindow } from '../app/utils';
+import config from '../config';
+import createTestStore from '../test/createTestStore';
+import loadOptimize from './loadOptimize';
+
+describe('loadOptimize', () => {
+  let store: Store;
+  let window: Window;
+  let dispatch: jest.SpyInstance;
+  const document = assertDocument();
+
+  beforeEach(() => {
+    store = createTestStore();
+    window = assertWindow();
+    dispatch = jest.spyOn(store, 'dispatch');
+
+    const createEvent = document.createEvent.bind(document);
+    const originalCreateElement = document.createElement.bind(document);
+
+    document.createElement = jest.fn((type: string) => {
+      const element = originalCreateElement(type);
+      const event = createEvent('Event');
+      event.initEvent('load');
+      setImmediate(() => element.onload && element.onload(event));
+
+      return element;
+    });
+  });
+
+  afterEach(() => {
+    document.head.innerHTML = '';
+    window.dataLayer = [];
+  });
+
+  it('injects correct <script> into head if in production', async() => {
+    config.DEPLOYED_ENV = 'openstax.org';
+    window.dataLayer.push = jest.fn();
```

Comment:
@TomWoodward added this to prevent registering sessions

I'm also wondering why overwriting `config.DEPLOYED_ENV` works above but `jest.mock('../config', () => ({DEPLOYED_ENV: 'openstax.org'}));` does not

Reply:
probably depends on the order of the importing and the mocking